server/src/main/java/org/opensearch/extensions/DiscoveryExtensionNode.java

```diff
@@ -89,6 +98,10 @@ public List<String> getImplementedInterfaces() {
         return implementedInterfaces;
     }
 
+    public Set<Scope> getScopes() {
```

Comment:
Those are the scopes the extensions needs, right? But in general, it may not get them, right?

Reply:
Yeah exactly.

Reply:
May be we could call them as `getRequiredScopes`, just to differentiate between scopes at runtime `getScopes()` and scopes at initialization time? What do you think?